tournamentcontrol/competition/tests/test_cross_division_matches.py

```diff
@@ -0,0 +1,216 @@
+        
+        form = MatchEditForm(instance=match, data=form_data)
+        self.assertTrue(form.is_valid(), form.errors)
+        
+        # Verify the flag is correctly set to False
+        saved_match = form.save()
+        self.assertFalse(saved_match.ignore_group_validation)
+
+    def test_cross_division_match_bypasses_validation(self):
+        """Test that matches with ignore_group_validation=True bypass division validation."""
+        match = factories.MatchFactory(
+            stage=self.stage1,
+            ignore_group_validation=True
+        )
+        
+        # Form should accept teams from different divisions
+        form_data = {
+            'home_team': self.team1_div1.pk,
+            'away_team': self.team1_div2.pk,  # Different division
+            'ignore_group_validation': '1',  # BooleanChoiceField expects "1" for True
+            'include_in_ladder': '1',  # BooleanChoiceField expects "1" for True
+        }
+        
+        form = MatchEditForm(instance=match, data=form_data)
+        self.assertTrue(form.is_valid(), form.errors)
+        
+        saved_match = form.save()
+        self.assertEqual(saved_match.home_team, self.team1_div1)
+        self.assertEqual(saved_match.away_team, self.team1_div2)
+        self.assertTrue(saved_match.ignore_group_validation)
+
+    def test_cross_division_match_form_includes_field(self):
+        """Test that MatchEditForm includes ignore_group_validation field."""
+        match = factories.MatchFactory(stage=self.stage1)
+        form = MatchEditForm(instance=match)
+        self.assertIn('ignore_group_validation', form.fields)
+        self.assertEqual(
+            form.fields['ignore_group_validation'].help_text,
+            "Allow this match to bypass strict Division and StageGroup validation. "
+            "Use for friendlies, showcase events, or cross-division matchups. "
+            "These matches will not count toward standings."
+        )
+
+    def test_same_team_validation_still_applies(self):
+        """Test that teams cannot play against themselves even with ignore_group_validation=True."""
+        match = factories.MatchFactory(
+            stage=self.stage1,
+            ignore_group_validation=True
+        )
+        
+        form_data = {
+            'home_team': self.team1_div1.pk,
+            'away_team': self.team1_div1.pk,  # Same team
+            'ignore_group_validation': '1',
+            'include_in_ladder': '1',
+        }
+        
+        form = MatchEditForm(instance=match, data=form_data)
+        self.assertFormError(form, 'away_team', ["Teams cannot be scheduled to play against itself."])
+
+    def test_stage_group_validation_bypassed_when_flag_set(self):
+        """Test that pool/stage group validation is bypassed when ignore_group_validation=True."""
+        # Create a stage with pools
+        pool1 = factories.StageGroupFactory(stage=self.stage1)
+        pool2 = factories.StageGroupFactory(stage=self.stage1)
```

Comment:
Check the syntax in how you're using the factories, it should be `create` method, on in some cases `create_batch` for multiple.